docs/concepts/custom-ui-headless-api.mdx

```diff
@@ -0,0 +1,662 @@
+---
+id: custom-ui-headless-api
+title: Integration Basics
+sidebar_label: Integration Basics
+toc_max_heading_level: 4
+---
+
+```mdx-code-block
+import Tabs from "@theme/Tabs"
+import TabItem from "@theme/TabItem"
+import CodeBlock from "@theme/CodeBlock"
+```
+
+Ory provides headless APIs for ease of integration. This ensures that Ory provides compatibility with many software stacks across
+different programming languages.
+
+This documents' focus is to provide concrete examples on custom UI integration. Flows are the fundamental of Ory's self-service
+APIs and are used to accomplish a variety of goals, such as `login`, `registration`, `settings`, `verification` and `recovery`.
+
+The self-service `flows` are standardized so that it follows the same steps regardless of which `flow` is integrated.
+
+A general overview of the flow can be summed up in two steps:
+
+1. Get or create the `flow` data, e.g. login, registration etc.
+2. Submit the `flow` with user data (e.g. username and password)
+
+For more information on the concepts, take a look at the [Self-service flow](../kratos/self-service) documentation.
+
+## Creating a flow
+
+All browser flows can be initialized with the optional `?return_to=<url>` query parameter. This parameter is used to redirect the
+user to the specified URL after the flow is successfully completed.
+
+Some flows, such as login have additional query parameters which can be used in both browser and native applications.
+
+## Reusing an existing flow
```

Comment:
I am not very happy with this section. What I think is confusing is:

- Wording: we are not "resuing" the flow, but rather fetching it. The reader might think they can create a flow and use it for multiple subsequent logins.
- The section should be further down when some of the more fundamental concepts are explained.
- The text is too generic. It is a complete mystery at this point why "Ory has initialized the flow and redirects to the UI". This might be more clear if that is explained in the "Create" section.